src/java.net.http/share/classes/jdk/internal/net/http/common/Deadline.java

```diff
@@ -266,10 +305,18 @@ static Deadline of(Instant instant) {
      * @param startInclusive  the start deadline, inclusive, not null
      * @param endExclusive  the end deadline, exclusive, not null
      * @return a {@code Duration}, not null
-     * @throws DateTimeException if the seconds between the deadline cannot be obtained
-     * @throws ArithmeticException if the calculation exceeds the capacity of {@code Duration}
      */
     public static Duration between(Deadline startInclusive, Deadline endExclusive) {
-        return Duration.between(startInclusive.deadline, endExclusive.deadline);
+        if (startInclusive.equals(endExclusive)) return Duration.ZERO;
+        try {
+            return Duration.between(startInclusive.deadline, endExclusive.deadline);
+        } catch (DateTimeException |                // "Instant exceeds minimum or maximum instant"
+                 ArithmeticException exception) {   // "long overflow"
+            // `Deadline` works with `Instant` under the hood.
+            // Delta between `Instant.MIN` and `Instant.MAX` fits in a `Duration`.
+            // Hence, we should never receive a numeric overflow while calculating the delta between two deadlines.
+            throw new IllegalStateException("Unexpected overflow", exception);
+        }
     }
+
```

Comment:
Do we need to change this method? I would just revert them.

Reply:
Oh - yes we don't take Temporal but Deadline as paramater which eliminates the possibility of overflow.